File: packages/pensopay_android/android/src/main/kotlin/com/flutter/pensopay/networking/pensopayapi/pensopaylink/subscriptions/PPGetSubscriptionRequest.kt
package com.flutter.pensopay.networking.pensopayapi.pensopaylink.subscriptions

import com.android.volley.Request
import com.flutter.pensopay.networking.pensopayapi.PPrequest
import com.flutter.pensopay.networking.pensopayapi.pensopaylink.models.PPSubscription

class PPGetSubscriptionRequest(subscription_id: Int): PPrequest<PPSubscription>(Request.Method.GET, "/subscription/$subscription_id", null, PPSubscription::class.java)

File: packages/pensopay_ios/ios/Classes/Networking/Pensopay API/Pensopay Link/Subscriptions/PPGetSubscriptionRequest.swift
//
//  PPGetSubscriptionRequest.swift
//  PensopaySDK
//
//  Created on 07/26/2022
//  Copyright © 2022 Pensopay. All rights reserved.
//

import Foundation

public class PPGetSubscriptionRequest : PPRequest {
    
    // MARK: - Properties
    
    public var subscription_id: Int
    
    
    // MARK: - Init
    
    public init(subscription_id: Int) {
        self.subscription_id = subscription_id
    }
    
    
    // MARK: - URL Request
    
    public func sendRequest(success: @escaping (_ result: PPSubscription) -> Void, failure: ((_ data: Data?, _ response: URLResponse?, _ error: Error?) -> Void)?) {
        guard let url = URL(string: "\(PensopayAPIBaseUrl)/subscriptions/\(self.subscription_id)") else {
            return
        }
        
        var request = URLRequest(url: url)
        request.httpMethod = "GET"
        request.setValue(headers.encodedAuthorization(), forHTTPHeaderField: "Authorization")
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        request.setValue("application/json", forHTTPHeaderField: "Accept")
        
        super.sendRequest(request: request, success: success, failure: failure)
    }
}
